Реализовываем анимацию количества призов

andikk opened the pull request:
---
:mortar_board: [Реализовываем анимацию количества призов](https://up.htmlacademy.ru/animation/1/user/643755/tasks/13)

:boom: https://htmlacademy-animation.github.io/643755-magic-vacation-1/12/

andikk committed:
таймер

andikk committed:
counter

sfi0zy commented on source/js/modules/counter.js:
```diff
@@ -0,0 +1,17 @@
+      clearInterval(myTimer);
+    }
+
+  }, 120);
```
:point_up: - Что значит магическое число 120?

:point_up: - Тормозилка fps - это универсальная штука, которая иногда пригождается в работе. Будет полезно сделать себе такую тормозилку, которая будет принимать в качестве параметра необходимое значение fps.

sfi0zy commented on source/js/modules/page.js:
```diff
@@ -8,6 +9,14 @@ export default () => {
+    startCounter(7, 1, 1, casesEl);
+
+    const codesEl = document.querySelector(`.prizes__item--codes b`);
+    startCounter(900, 11, 60, codesEl);
```
:point_up: - Что значат магические числа здесь?

:point_up: - Иногда, если уж нужно передавать куда-то много странных параметров, то можно сделать объект с параметрами, где у каждого будет имя. Код станет самозадокументированным, если это там можно сказать.

sfi0zy reviewed (commented):
:pencil2: **Первичная проверка.**

**Обозначения:**

:red_circle: - Нужно обязательно исправить.
:point_up: - Нужно подумать над этим, принять к сведению.
:watch: - Нужно подождать (что-то пошло не так, вы ни при чем).
:heavy_check_mark: - Задание принято.
:x: - Задание не принято.
:large_blue_diamond: - Важное сообщение.

**Чеклист:**

- [ ] Частота кадров уменьшена до 12 для снижения нагрузки на процессор.
- [ ] Длительность анимации одного числа - меньше секунды.
- [x] Число 900 начинает отсчет с 11.

**Комментарии:**

:red_circle: - Третий таймер заканчивает счет на 911, а не на 900.

:red_circle: - Длительность анимации у третьего таймера явно больше, чем на видео концепте.

:point_up: - По магическому числу 120 и тому, что анимация заканчивается не там, где должна, могу предположить, что вы слегка запутались в арифметике. Это распространенная проблема в нашей сфере, обычно связанная с некоторой невнимательностью, но иногда, как здесь, анимации начинают выглядеть не так приятно, как задумывались. Постарайтесь разобраться с тем, что здесь пошло не так. И в при выполнении будущих заданий тоже обращайте на такие места внимание. Вы же определенно можете такое сразу делать хорошо.

**Итог:**

:x: - Выполнены не все требования ТЗ. Необходима доработка.

andikk committed:
css

andikk committed:
fps проверял по такой аналогии:
var lastLoop = new Date();
function gameLoop() {
    var thisLoop = new Date();
    var fps = 1000 / (thisLoop - lastLoop);
    lastLoop = thisLoop;
    ...
}

получалось чуть меньше 12. Хотя я не уверен, что правильно реализовал торможение fps.

Я гуглил в интернете, находил готовые функции что-то типа http://jsfiddle.net/m1erickson/CtsY3/ но так и не смог реализовать это у себя.

andikk committed:
fps проверял по такой аналогии:
var lastLoop = new Date();
function gameLoop() {
    var thisLoop = new Date();
    var fps = 1000 / (thisLoop - lastLoop);
    lastLoop = thisLoop;
    ...
}

получалось чуть меньше 12. Хотя я не уверен, что правильно реализовал торможение fps.

Я гуглил в интернете, находил готовые функции что-то типа http://jsfiddle.net/m1erickson/CtsY3/ но так и не смог реализовать это у себя.

sfi0zy commented on source/js/modules/counter.js:
```diff
@@ -1,17 +1,20 @@
-    if (counter >= maxCount) {
-      clearInterval(myTimer);
-    }
+  }, 1000 / fps);
```
:point_up: - Посоветовал бы все же разобраться с тем вариантом, который дается в учебнике. Он может быть более универсальным и надежным в контексте сложных задач, когда мы сталкиваемся с нехваткой вычислительной мощности у клиента. SetInterval может дать эффект "интервалов больше, чем мы реально можем отрисовать".

:point_up: - Но вообще производительность - это отдельная тема, особенно по части WebGL, но к этому мы потом еще придем.

sfi0zy reviewed (commented):
:pencil2: **Повторная проверка.**

**Чеклист:**

- [x] Частота кадров уменьшена до 12 для снижения нагрузки на процессор.
- [x] Длительность анимации одного числа - меньше секунды.
- [x] Число 900 начинает отсчет с 11.

**Комментарии:**

:+1: - Теперь выглядит гораздо лучше.

> Я гуглил в интернете, находил готовые функции что-то типа http://jsfiddle.net/m1erickson/CtsY3/ но так и не смог реализовать это у себя.

У вас же вроде даже в учебнике был такой пример. Мне сложно дать комментарий по существу, т.к. "не смог реализовать" - это слишком общее описание проблемы, учитывая наличие рабочего примера.

**Итог:**

:heavy_check_mark: :+1: - Выполнены все требования ТЗ, анимация соответствует видео с концептом проекта. Задание принято.
